openedx/features/course_experience/utils.py

```diff
@@ -265,10 +266,12 @@ def reset_deadlines_banner_should_display(course_key, request):
     is_course_staff = bool(
         request.user and course_overview and has_access(request.user, 'staff', course_overview, course_overview.id)
     )
-    if is_self_paced and (not is_course_staff) and (not course_end_date or timezone.now() < course_end_date):
-        if (CourseEnrollment.objects.filter(
-            course=course_overview, user=request.user, mode=CourseMode.VERIFIED
-        ).exists()):
+    if is_self_paced and (is_course_staff) and (not course_end_date or timezone.now() < course_end_date):
+        if CourseEnrollment.objects.filter(
+            course=course_overview, user=request.user,
+        ).filter(
+            Q(mode=CourseMode.AUDIT) | Q(mode=CourseMode.VERIFIED)
+        ).exists():
```

Comment:
Because of this change, and since we use this method in other areas of the app (course outline), Im actually thinking now maybe the change to the banner on the course outline page should be done in this same PR